timemachine/ff/handlers/nonbonded.py

```diff
@@ -187,26 +189,29 @@ def bond_smirks_matches(mol, smirks_list):
     * Order within each smirks pattern matters
         For example, "[#6:1]~[#1:2]" and "[#1:1]~[#6:2]" will match atom pairs in the opposite order
     """
-    oemol = convert_to_oe(mol)
-    AromaticityModel.assign(oemol)
+    if not mol.HasProp(BOND_SMIRK_MATCH_CACHE):
+        oemol = convert_to_oe(mol)
+        AromaticityModel.assign(oemol)
 
-    bond_idxs = []  # [B, 2]
-    type_idxs = []  # [B]
+        bond_idxs = []  # [B, 2]
+        type_idxs = []  # [B]
 
-    for type_idx, smirks in enumerate(smirks_list):
-        matches = oe_match_smirks(smirks, oemol)
+        for type_idx, smirks in enumerate(smirks_list):
+            matches = oe_match_smirks(smirks, oemol)
 
-        for matched_indices in matches:
-            a, b = matched_indices[0], matched_indices[1]
-            forward_matched_bond = [a, b]
-            reverse_matched_bond = [b, a]
+            for matched_indices in matches:
+                a, b = matched_indices[0], matched_indices[1]
+                forward_matched_bond = [a, b]
+                reverse_matched_bond = [b, a]
 
-            already_assigned = forward_matched_bond in bond_idxs or reverse_matched_bond in bond_idxs
-
-            if not already_assigned:
-                bond_idxs.append(forward_matched_bond)
-                type_idxs.append(type_idx)
+                already_assigned = forward_matched_bond in bond_idxs or reverse_matched_bond in bond_idxs
 
+                if not already_assigned:
+                    bond_idxs.append(forward_matched_bond)
+                    type_idxs.append(type_idx)
```

Comment:
nit: might be nice to split this into a separate function `compute_bond_smirks_matches` to avoid a large indented block